clippy_lints/src/mut_mutex_lock.rs

```diff
@@ -0,0 +1,64 @@
+use crate::utils::{is_type_diagnostic_item, span_lint_and_help};
+use if_chain::if_chain;
+use rustc_hir::{Expr, ExprKind, Mutability};
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_middle::ty;
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+
+declare_clippy_lint! {
+    /// **What it does:** Checks for `&mut Mutex::lock` calls
+    ///
+    /// **Why is this bad?** `Mutex::lock` is less efficient than
+    /// calling `Mutex::get_mut`
```

Comment:
From the documentation:

> Since this call borrows the `Mutex` mutably, no actual locking needs to take place -- the mutable borrow statically guarantees no locks exist.

In addition you also have a statically guarantee that the mutex isn't locked, instead of just a runtime guarantee.

Reply:
Added that sentence to the description.

Reply:
Added that sentence to the description.